tests/e2e/initialization/init.Dockerfile

```diff
@@ -22,14 +24,22 @@ RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep f6282df732a13dec836cda1f399dd
 # CosmWasm: copy the right library according to architecture. The final location will be found by the linker flag `-lwasmvm_muslc`
 RUN cp /lib/libwasmvm_muslc.$(uname -m).a /lib/libwasmvm_muslc.a
 
-RUN BUILD_TAGS=muslc LINK_STATICALLY=true make build-e2e-chain-init
+RUN BUILD_TAGS=muslc LINK_STATICALLY=true E2E_SCRIPT_NAME=${E2E_SCRIPT_NAME} make build-e2e-script
 
 ## Deploy image
 FROM ubuntu
 
-COPY --from=build /osmosis/build/chain_init /bin/chain_init
+# Args only last for a single build stage - renew
+ARG E2E_SCRIPT_NAME
+
+COPY --from=build /osmosis/build/${E2E_SCRIPT_NAME} /bin/${E2E_SCRIPT_NAME}
 
 ENV HOME /osmosis
 WORKDIR $HOME
 
-ENTRYPOINT [ "chain_init" ]
+# Docker ARGs are not expanded in ENTRYPOINT in the exec mode. At the same time,
+# it is impossible to add CMD arguments when running a container in the shell mode.
+# As a workaround, we create the entrypoint.sh script to bypass these issues.
+RUN echo "#!/bin/bash\n${E2E_SCRIPT_NAME} \"\$@\"" >> entrypoint.sh && chmod +x entrypoint.sh
```

Comment:
For my learning, what are the semantics here `\"\$@\""`? I have never seen this before.

Reply:
This is bash syntax for expanding cmd line parameters.  

Basically, since we want to call chain init and node init scripts by providing them with certain initialization arguments via Docker, we need to be able to expand them in the bash script.

This post points to a relevant man page about the syntax:
https://stackoverflow.com/questions/3898665/what-is-in-bash